shell/src/main/java/alluxio/cli/bundler/InfoCollectorAll.java

```diff
@@ -0,0 +1,249 @@
+
+    // Execute the Collectors one by one
+    // Reduce the RPC retry max duration to fail earlier for CLIs
+    conf.set(PropertyKey.USER_RPC_RETRY_MAX_DURATION, "5s", Source.DEFAULT);
+    InfoCollectorAll shellAll = new InfoCollectorAll(conf);
+
+    // Validate commands
+    if (argv.length < 2) {
+      shellAll.printUsage();
+      System.exit(-1);
+    }
+    String action = argv[0];
+    String targetDir = argv[1];
+
+    List<String> allHosts = new ArrayList<>(shellAll.getHosts());
+    System.out.println(String.format("Init thread pool for %s hosts", allHosts.size()));
+    shellAll.mExecutor = Executors.newFixedThreadPool(allHosts.size());
+
+    // Invoke infoBundle on each host
+    List<CompletableFuture<CommandReturn>> futureList = new ArrayList<>();
+    for (String host : allHosts) {
+      System.out.println(String.format("Execute InfoCollector on host %s", host));
+
+      CompletableFuture<CommandReturn> future = CompletableFuture.supplyAsync(() -> {
+        String workDir = conf.get(PropertyKey.WORK_DIR);
+        String alluxioBinPath = Paths.get(workDir, "bin/alluxio")
+                .toAbsolutePath().toString();
+        System.out.println(String.format("host: %s, alluxio path %s", host, alluxioBinPath));
+
+        try {
+          String[] infoBundleArgs =
+                  (String[]) ArrayUtils.addAll(new String[]{alluxioBinPath, "infoBundle"}, argv);
+          CommandReturn cr = ShellUtils.sshExecCommandWithOutput(host, infoBundleArgs);
+          return cr;
+        } catch (Exception e) {
+          // TODO(jiacheng): What exceptions can be thrown here?
+          LOG.error("Execution failed %s", e);
+          return new CommandReturn(1, e.toString());
+        }
+      }, shellAll.mExecutor);
+      futureList.add(future);
+      System.out.println(String.format("Invoked infoBundle command on host %s", host));
+    }
+
+    // Collect the execution results
+    List<CommandReturn> results;
+    try {
+      results = all(futureList).get();
+      System.out.println(String.format("%s results completed", results.size()));
+    } catch (InterruptedException | ExecutionException e) {
+      System.out.println("Command failed!");
+      throw new IOException(e);
+    }
+
+    // Inspect all command results
+    if (results.size() != allHosts.size()) {
+      System.out.println("Host size mismatch!");
+    } else {
+      for (int i = 0; i < allHosts.size(); i++) {
+        CommandReturn c = results.get(i);
+        String host = allHosts.get(i);
+        if (c.getExitCode() != 0) {
+          System.out.println(String.format("Command failed on host %s", host));
+          System.out.println(c.getFormattedOutput());
+        } else {
+          System.out.println(String.format("Success on host %s", host));
+        }
+      }
+    }
+
+    // Collect all tarballs to local
+    File tempDir = AlluxioTestDirectory.createTemporaryDirectory(TEMP_DIR_NAME);
```

Comment:
It is a little strange to create a temporary directory named with `alluxio-tests`. You can probably use guava's ` Files.createTempDir()` instead.